src/unix/freebsd.c

```diff
@@ -196,12 +196,20 @@ int uv_set_process_title(const char* title) {
 
 
 int uv_get_process_title(char* buffer, size_t size) {
+  size_t len;
+
+  if (buffer == NULL || size == 0)
+    return -EINVAL;
+
   if (process_title) {
-    strncpy(buffer, process_title, size);
+    len = strlen(process_title) + 1;
+
+    if (size < len)
+      return -ENOBUFS;
+
+    memcpy(buffer, process_title, len);
```

Comment:
Oh. If it wasn't because of synchronization, all but one of the implementations relies on `strlen()`, so the null could be safely assumed.